src/Relation/Binary/Lattice/Structures.agda

```diff
@@ -0,0 +1,204 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Structures for order-theoretic lattices
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via `Relation.Binary.Lattice`.
+
+{-# OPTIONS --without-K --safe #-}
+
+module Relation.Binary.Lattice.Structures where
+
+open import Algebra.Core
+open import Algebra.Definitions
+open import Data.Product using (_×_; _,_)
+open import Level using (suc; _⊔_)
+open import Relation.Binary
+open import Relation.Binary.Lattice.Definitions
+
+------------------------------------------------------------------------
+-- Join semilattices
+
+record IsJoinSemilattice {a ℓ₁ ℓ₂} {A : Set a}
+                         (_≈_ : Rel A ℓ₁) -- The underlying equality.
+                         (_≤_ : Rel A ℓ₂) -- The partial order.
```

Comment:
So one of the advantages of this refactor is that we can lift 
```
{a ℓ₁ ℓ₂} {A : Set a}
(_≈_ : Rel A ℓ₁) -- The underlying equality.
(_≤_ : Rel A ℓ₂) -- The partial order.
```
to be module arguments!

Reply:
@MatthewDaggitt So it will be like [I have skipped comments here].
`module Relation.Binary.Lattice.Structures {a ℓ₁ ℓ₂} {A : Set a} (_≈_ : Rel A ℓ₁) (_≤_ : Rel A ℓ₂) where`

 `record IsJoinSemilattice (_∨_ : Op₂ A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where ....`
`record IsLattice (_∨_ : Op₂ A)  (_∧_ : Op₂ A) Set (a ⊔ ℓ₁ ⊔ ℓ₂) where ... `

and
 ` isJoinSemilattice : IsJoinSemilattice _∨_` ?

Reply:
Yes, much tidier right?

Reply:
Yes. I will push the changes :)